src/BenchmarkDotNet/Exporters/MarkdownExporter.cs

```diff
@@ -85,6 +87,7 @@ public enum MarkdownHighlightStrategy
         [PublicAPI] protected bool ColumnsStartWithSeparator;
         [PublicAPI] protected string BoldMarkupFormat = "**{0}**";
         [PublicAPI] protected bool EscapeHtml;
+        [PublicAPI] protected bool EscapePipe;
```

Comment:
I can see that you are not setting the value of this property to `true` for `StackOverflow`, but for some reason the SO exporter is escaping it:

```cs
[MarkdownExporterAttribute.Default]
[MarkdownExporterAttribute.GitHub]
[MarkdownExporterAttribute.StackOverflow]
[MarkdownExporterAttribute.Atlassian]
public class Intro_MethodsParamsEscaped
{
    [Params("Should|Escape", "ShouldNotEscape")] public string Param;

    [Benchmark] public bool Method() => Param is null;
}
```

```cmd
dotnet run -c Release -f net5.0 --filter *Intro_MethodsParamsEscaped* --job dry
```

![image](https://user-images.githubusercontent.com/6011991/151505899-9d5a56a5-34d4-4979-8d9e-3878e25e7288.png)

Could you please ensure that it does not get escaped for the SO exporter?

Reply:
Ok, this should be fixed now.

Do we want to apply this escaping to console output?  It would render as ```\|``` in the console, which is probably not desired, however there are comments in the code indicating that users often copy-paste from the console into Github:https://github.com/dotnet/BenchmarkDotNet/blob/67bf3c0a170d1c4270420bc122fbf795f1b3546c/src/BenchmarkDotNet/Exporters/MarkdownExporter.cs#L37